### PR TITLE
Add new `feature` category for changelog entries

### PR DESCRIPTION
**Pull request by cdce8p**
## Description
At the moment, there isn't a good category for new features which aren't checks.
The next best option would be `other` but that doesn't seem quite right.

Refs: https://github.com/PyCQA/pylint/issues/7362

### CODE REVIEW

**Review by Pierre-Sassoulas — DISMISSED** [review]
I think it makes sense, I also noticed that 👍

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -5,9 +5,9 @@ To ease the process of reviewing your PR, do make sure to complete the following
 
 - [ ] Write a good description on what the PR does.
 - [ ] Create a news fragment with `towncrier create <IssueNumber>.<type>` which will be
-  included in the changelog. `<type>` can be one of: breaking, user_action, new_check, removed_check, extension,
-  false_positive, false_negative, bugfix, other, internal. If necessary you can write
-  details or offer examples on how the new change is supposed to work.
+  included in the changelog. `<type>` can be one of: breaking, user_action, feature,
+  new_check, removed_check, extension, false_positive, false_negative, bugfix, other, internal.
```
Comment (DudeNr33) [inline .github/PULL_REQUEST_TEMPLATE.md:9]: This section is also in [contribute.rst](https://github.com/PyCQA/pylint/blob/main/doc/development_guide/contributor_guide/contribute.rst), so we'd need to add this there as well.

**Review by DudeNr33 — CHANGES_REQUESTED** [review]
Looks good to me, when `contribute.rst` is updated as well it can be merged. 👍

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by Pierre-Sassoulas** [comment]
Sorry @DudeNr33 I was forced to re-request review to be able to merge :)

**Comment by cdce8p** [comment]
> Sorry @DudeNr33 I was forced to re-request review to be able to merge :)

@Pierre-Sassoulas Usually we should be able to dismiss the review. I don't seem to be able to either though 🤔
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/dismissing-a-pull-request-review

**Comment by Pierre-Sassoulas** [comment]
>  I don't seem to be able to either though 

Yeah, same here, I don't know why.

**Comment by cdce8p** [comment]
> > I don't seem to be able to either though
> 
> Yeah, same here, I don't know why.

From the Github [docs](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#require-pull-request-reviews-before-merging)

> Optionally, you can restrict the ability to dismiss pull request reviews to specific people or teams. For more information, see "[Dismissing a pull request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/dismissing-a-pull-request-review)."

--
Just FYI I enabled `Always suggest updating pull request branches` in the repo settings to suggest the `Update branch` button. Usually with towncrier we shouldn't need to use it but it can be helpful sometimes anyway.

**Review by DudeNr33 — APPROVED** [review]
Sorry for the delay!

**Pull request merged**